nixos/modules/services/home-automation/homebridge.nix

```diff
@@ -0,0 +1,756 @@
+              ${optionalString (
+                cfg.bridge.manufacturer != null
+              ) '',"manufacturer": "${cfg.bridge.manufacturer}"''}
+              ${optionalString (cfg.bridge.model != null) '',"model": "${cfg.bridge.model}"''}
+              ${optionalString (
+                cfg.bridge.disableIpc != null
+              ) '',"disableIpc": ${toString cfg.bridge.disableIpc}''}
+            } |
+            .platforms |= map(
+              if .platform == "config" then
+                . * {
+                  port: ${toString cfg.ui.port},
+                  standalone: true,
+                  restart: "${restartCommand}",
+                  homebridgePackagePath: "${homebridgePackagePath}",
+                  sudo: false,
+                  log: {
+                    method: "systemd",
+                    service: "homebridge"
+                  }
+                  ${optionalString (cfg.ui.host != null) '',"host": "${cfg.ui.host}"''}
+                  ${optionalString (cfg.ui.proxyHost != null) '',"proxyHost": "${cfg.ui.proxyHost}"''}
+                  ${optionalString (cfg.ui.auth != null) '',"auth": "${cfg.ui.auth}"''}
+                  ${optionalString (
+                    cfg.ui.sessionTimeout != null
+                  ) '',"sessionTimeout": ${toString cfg.ui.sessionTimeout}''}
+                  ${optionalString (cfg.ui.ssl != null) ''
+                    ,"ssl": {
+                      "key": "${cfg.ui.ssl.key}",
+                      "cert": "${cfg.ui.ssl.cert}",
+                      "pfx": "${cfg.ui.ssl.pfx}",
+                      "passphrase": "${cfg.ui.ssl.passphrase}"
+                    }''}
+                  ${optionalString (cfg.ui.tempUnits != null) '',"tempUnits": "${cfg.ui.tempUnits}"''}
+                  ${optionalString (cfg.ui.theme != null) '',"theme": "${cfg.ui.theme}"''}
+                  ${optionalString (cfg.ui.lang != null) '',"lang": "${cfg.ui.lang}"''}
+                  ${optionalString (
+                    cfg.ui.loginWallpaper != null
+                  ) '',"loginWallpaper": "${cfg.ui.loginWallpaper}"''}
+                  ${optionalString (
+                    cfg.ui.scheduledBackupDisable != null
+                  ) '',"scheduledBackupDisable": ${toString cfg.ui.scheduledBackupDisable}''}
+                  ${optionalString (
+                    cfg.ui.scheduledBackupPath != null
+                  ) '',"scheduledBackupPath": "${cfg.ui.scheduledBackupPath}"''}
+                  ${optionalString (cfg.ui.debug != null) '',"debug": ${toString cfg.ui.debug}''}
+                  ${optionalString (
+                    cfg.ui.disableServerMetricsMonitoring != null
+                  ) '',"disableServerMetricsMonitoring": ${toString cfg.ui.disableServerMetricsMonitoring}''}
+                }
+              else
+                .
+              end
+            )
+        EOF
+          )
```

Comment:
If you do my `freeformType` suggestion, this is also now completely unnecessary. Just use the json writer, and then keep your later config merger to merge the two files.